src/backend/distributed/operations/shard_transfer.c

```diff
@@ -1839,27 +1897,22 @@ RecreateTableDDLCommandList(Oid relationId)
 
 
 /*
- * MarkForDropColocatedShardPlacement marks the shard placement metadata for
- * the given shard placement to be deleted in pg_dist_placement. The function
- * does this for all colocated placements.
+ * DropColocatedShardPlacements drops the shard placement metadata for
+ * the shard placements of given colocated shard interval list from pg_dist_placement.
  */
 static void
-MarkForDropColocatedShardPlacement(ShardInterval *shardInterval, char *nodeName,
-								   int32 nodePort)
+DropColocatedShardPlacements(List *colocatedShardList,
```

Comment:
nit: This function doesn't call ColocatedShardIntervalList anymore. So there's no reason why it should have colocated in its name or arguments (or comment). It simple deletes all placement rows for the given nodes for the given shard list, no matter if they are colocated or not. Also maybe to make clearer that this only deletes the placement from the metadata and not from the actual node, we could include something like `FromMetadata` in the function name.
```suggestion
DropShardPlacementsFromMetadata(List *shardList,
```